ingest/api/model.py

```diff
@@ -0,0 +1,316 @@
+            "https://www.iso.org/iso-8601-date-and-time-format.html"
+        ),
+    )
+    function: str = Field(
+        ...,
+        description=(
+            "Function used on the data during the aggregation period."
+            "Must be one of the functions given in CF cell_methods table."
+            "https://cfconventions.org/Data/cf-conventions/cf-conventions-1.7/build/ape.html"
+        ),
+    )
+    history: Optional[str] = Field(
+        None,
+        description=(
+            "Provides an audit trail for modifications to the original data. "
+            "This attribute is also in the NetCDF Users Guide: 'This is a character array with a line for each "
+            "invocation of a program that has modified the dataset. "
+            "Well-behaved generic netCDF applications should append a line containing: date, time of day, user name, "
+            "program name and command arguments.' To include a more complete description you can append a reference to"
+            " an ISO Lineage entity; see NOAA EDM ISO Lineage guidance."
+        ),
+    )
+    datetime: str = Field(
+        ...,
+        description="Identifies the date/time of the datas being published, in RFC3339 format.",
+    )
+    start_datetime: Optional[str] = Field(
+        None,
+        description="Identifies the start date/time date of the data being published, in RFC3339 format.",
+    )
+    end_datetime: Optional[str] = Field(
+        None,
+        description="Identifies the end date/time date of the data being published, in RFC3339 format.",
+    )
+    processing_level: Optional[str] = Field(
+        None,
+        description="A textual description of the processing (or quality control) level of the data.",
+    )
+    content: Optional[Content] = Field(None, description="Actual data content")
+    integrity: Optional[Integrity] = Field(
+        None, description="Specifies a checksum to be applied to the data to ensure that the download is accurate."
+    )
+
+    @model_validator(mode="after")
+    def check_datetime_iso(self) -> "Properties":
+        try:
+            datetime.strptime(self.datetime, "%Y-%m-%dT%H:%M:%S.%f%z")
+        except Exception:
+            raise ValueError(f"{self.datetime} not in ISO format(YYYY-MM-DDTHH:MM:SS.ssssss)")
+        return self
+
+
+class Link(BaseModel):
+    href: str = Field(..., example="http://data.example.com/buildings/123")
+    rel: str = Field(..., example="alternate")
+    type: Optional[str] = Field(None, example="application/geo+json")
+    hreflang: Optional[str] = Field(None, example="en")
+    title: Optional[str] = Field(None, example="Trierer Strasse 70, 53115 Bonn")
+    length: Optional[int] = None
+
+
+class JsonMessageSchema(BaseModel):
+    type: Type
+    geometry: Union[Geometry, Geometry1]
+    properties: Properties
+    links: List[Link] = Field(..., min_items=1)
+    version: str
+
+    def dict(self, *args, **kwargs) -> Dict[str, Any]:
```

Comment:
Why do you need this method?
If you need the dict from a model use the model_dump method inherited from BaseModel

Reply:
for nested objects to a dictionary, we need to customize the serialization process